db/migrate/20220228173251_remove_drpm_from_ignorable_content.rb

```diff
@@ -1,7 +1,11 @@
 class RemoveDrpmFromIgnorableContent < ActiveRecord::Migration[6.0]
   def up
-    Katello::RootRepository.select { |r| r&.ignorable_content&.include? "drpm" }.each do |root|
-      root.ignorable_content = root.ignorable_content - ["drpm"]
+    Katello::RootRepository.select { |r| !r&.ignorable_content&.blank? }.each do |root|
+      if root.ignorable_content.include? "srpm"
```

Comment:
I got a NilClass error one of the times I tried to run this. Better to be safe..

```suggestion
      if root&.ignorable_content&.include? "srpm"
```

Reply:
Ack..Will update.. 😸

Reply:
Wait..given the outer loop `Katello::RootRepository.select { |r| !r&.ignorable_content&.blank? }.each do |root|`, I don't understand how it'd enter the loop for nil root..

Reply:
```
NoMethodError: undefined method `include?' for nil:NilClass
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:870:in `public_send'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:870:in `exec_migration'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:851:in `block (2 levels) in migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:850:in `block in migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:849:in `migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1037:in `migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1329:in `block in execute_migration_in_transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1380:in `block in ddl_transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `block in transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:319:in `block in within_new_transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/transactions.rb:209:in `transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1380:in `ddl_transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1328:in `execute_migration_in_transaction'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1302:in `each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1302:in `migrate_without_lock'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1251:in `block in migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1401:in `block in with_advisory_lock'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1416:in `block in with_advisory_lock_connection'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1416:in `with_advisory_lock_connection'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1397:in `with_advisory_lock'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1251:in `migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1086:in `up'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/migration.rb:1061:in `migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/tasks/database_tasks.rb:237:in `migrate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/railties/databases.rake:92:in `block (3 levels) in <main>'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/railties/databases.rake:90:in `each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.1.7/lib/active_record/railties/databases.rake:90:in `block (2 levels) in <main>'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/railties-6.1.7/lib/rails/commands/rake/rake_command.rb:24:in `block (2 levels) in perform'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/railties-6.1.7/lib/rails/commands/rake/rake_command.rb:24:in `block in perform'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/rake-13.0.6/lib/rake/rake_module.rb:59:in `with_application'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/railties-6.1.7/lib/rails/commands/rake/rake_command.rb:18:in `perform'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/railties-6.1.7/lib/rails/command.rb:50:in `invoke'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/railties-6.1.7/lib/rails/commands.rb:18:in `<main>'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/bootsnap-1.15.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/bootsnap-1.15.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `block in require'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `require'
/home/vagrant/foreman/bin/rails:9:in `<top (required)>'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/lib/spring/client/rails.rb:28:in `load'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/lib/spring/client/rails.rb:28:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/lib/spring/client/command.rb:7:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/lib/spring/client.rb:30:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/bin/spring:49:in `<top (required)>'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/lib/spring/binstub.rb:11:in `load'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/spring-2.1.1/lib/spring/binstub.rb:11:in `<top (required)>'
/home/vagrant/foreman/bin/spring:13:in `require'
/home/vagrant/foreman/bin/spring:13:in `<top (required)>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
[vagrant@centos8-katello-devel-stable foreman]$ bundle exec rails db:migrate
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:68: warning: already initialized constant Net::ProtocRetryError
/usr/share/ruby/net/protocol.rb:66: warning: previous definition of ProtocRetryError was here
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:214: warning: already initialized constant Net::BufferedIO::BUFSIZE
/usr/share/ruby/net/protocol.rb:206: warning: previous definition of BUFSIZE was here
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:541: warning: already initialized constant Net::NetPrivate::Socket
/usr/share/ruby/net/protocol.rb:503: warning: previous definition of Socket was here
2022-12-21T15:09:54 [I|app|] Rails cache backend: File
2022-12-21T15:09:55 [W|app|] You are trying to replace import_subnets from . Adding allowed actions from plugin permissions to the existing one.
2022-12-21T15:09:55 [W|app|] DEPRECATION WARNING: You are using a deprecated behavior, it will be removed in version 3.3, initial value of setting 'instance_id' should be created in a migration (called from block in load_definitions at /home/vagrant/foreman/app/services/setting_registry.rb:131)
2022-12-21T15:09:55 [D|app|] Registering 4 assets for plugin foreman-tasks precompilation
2022-12-21T15:09:55 [W|app|] DEPRECATION WARNING: Initialization autoloaded the constants Net::Validations, SettingRegistry, HiddenValue, SettingPresenter, Foreman::Deprecation, SettingSelectCollection, MediumProviders, MediumProviders::Provider, MediumProviders::Default, Foreman::Telemet
```

Reply:
Interesting..I guess the `!r&.ignorable_content&.blank?` check isn't checking nil when it should. Not sure..I will play around a little.

Reply:
Oh well.. 🤦🏼 
`
2] pry(main)> !nil&.blank?
=> true

`

Reply:
Updated to check for root.ignorable_content before calling includes.